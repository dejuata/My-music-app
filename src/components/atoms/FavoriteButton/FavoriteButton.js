import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { FaHeart } from 'react-icons/fa';

export const FavoriteButton = ({ funcActive, funcDeactive, state = false}) => {

    const [active, setActive] = useState(state)

    const handleOnClick = () => {
        !active ? funcActive() : funcDeactive()
        setActive(!active)
    }

    return (
        <button
            className='a-favorite-button'
            onClick={ handleOnClick }
        >
            <FaHeart
                className={` a-favorite__icon ${ active && 'a-favorite__icon--active'}`}
            />
        </button>
    )
}

FavoriteButton.propTypes = {
    funcActive: PropTypes.func.isRequired,
    funcDeactive: PropTypes.func.isRequired,
    state: PropTypes.bool
}