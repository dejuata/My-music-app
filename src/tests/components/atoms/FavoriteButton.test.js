import React from 'react';
import { shallow } from 'enzyme';
import { FavoriteButton } from '../../../components/atoms/FavoriteButton/FavoriteButton';

describe('Test FavoriteButton', () => {

    const handleActive = jest.fn();

    const handleDeactive = jest.fn();

    const state = false;

    const wrapper = shallow(
        <FavoriteButton
            funcActive={ handleActive }
            funcDeactive={ handleDeactive }
            state={ state }
        />
    )


    test('Should be render FavoriteButton', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('Should be change the state to true', () => {

        wrapper.find('button').simulate('click');
        expect( wrapper.find('FaHeart').hasClass('a-favorite__icon--active') ).toBe(true)

    })

    test('Should be change the state to false', () => {

        wrapper.find('button').simulate('click');
        expect( wrapper.find('FaHeart').hasClass('a-favorite__icon--active') ).toBe(false)

    })

    test('Should be execute handleDeactive', () => {
        const wrapper = shallow(
            <FavoriteButton
                funcActive={ handleActive }
                funcDeactive={ handleDeactive }
                state={ true }
            />
        )

        wrapper.find('button').simulate('click');
        expect( handleDeactive ).toHaveBeenCalledTimes(1);
    })

    test('Should be execute handleActive', () => {
        const wrapper = shallow(
            <FavoriteButton
                funcActive={ handleActive }
                funcDeactive={ handleDeactive }
                state={ false }
            />
        )

        wrapper.find('button').simulate('click');
        expect( handleActive ).toHaveBeenCalledTimes(1);
    })


})