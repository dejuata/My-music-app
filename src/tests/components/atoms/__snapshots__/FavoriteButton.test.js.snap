// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FavoriteButton Should be render FavoriteButton 1`] = `
<button
  className="a-favorite-button"
  onClick={[Function]}
>
  <FaHeart
    className=" a-favorite__icon false"
  />
</button>
`;
